src/array/cuda/spmat_op_impl_csr.cu

```diff
@@ -464,12 +462,44 @@ CSRMatrix CSRSliceMatrix(CSRMatrix csr, runtime::NDArray rows, runtime::NDArray
   IdArray mask = Full(0, csr.indices->shape[0], nbits, ctx);
   // A count for how many masked values per row.
   IdArray count = NewIdArray(csr.num_rows, ctx, nbits);
-  const int nt = cuda::FindNumThreads(csr.num_rows);
-  const int nb = (csr.num_rows + nt - 1) / nt;
+  CUDA_CALL(cudaMemset(count.Ptr<IdType>(), 0, sizeof(IdType) * (csr.num_rows)));
+
+  const int64_t nnz_csr = csr.indices->shape[0];
+  const int nt = 256;
+
+  // In general ``cols'' array is sorted. But it is not guaranteed.
+  // Hence checking and sorting array first. Sorting is not in place.
+  auto device = runtime::DeviceAPI::Get(ctx);
+  auto cols_size = cols->shape[0];
+  int8_t* flags = static_cast<int8_t*>(device->AllocWorkspace(ctx, cols_size-1));
+  int nb = (cols_size + nt - 1) / nt;
+  CUDA_KERNEL_CALL(_CheckSorted,
+      nb, nt, 0, thr_entry->stream,
+      cols.Ptr<IdType>(), cols_size, flags);
+  bool is_sorted = dgl::cuda::AllTrue(flags, cols_size-1, ctx);
+  device->FreeWorkspace(ctx, flags);
+
+  IdType *sorted_cols = cols.Ptr<IdType>();
+  if (is_sorted == false) {
```

Comment:
If we are going to have separate code paths for when columns is sorted or not, I think we need add a test for unsorted columns in here https://github.com/dmlc/dgl/blob/master/tests/cpp/test_spmat_csr.cc#L541.

Reply:
Based on our discussions and observing that sorting does not bring more than 10% overhead, I removed that check to simplify the implementation.